projects/packages/jetpack-mu-wpcom/src/features/domain-email-nag/domain-email-nag.php

```diff
@@ -32,52 +32,80 @@ function get_account_url() {
+ * Get the current domain for the site.
  */
-function get_domain_with_unverified_email() {
-	if ( ! class_exists( 'Domain_Management' ) ) {
-		return false;
-	}
-
-	$domains = \Domain_Management::get_paid_domains_with_icann_verification_status();
+function get_domain() {
+	$urlparts = wp_parse_url( home_url() );
+	return $urlparts['host'];
+}
 
-	foreach ( $domains as $domain ) {
-		if ( $domain['is_pending_icann_verification'] === true ) {
-			return $domain['domain'];
-		}
-	}
-	return false;
+/**
+ * Checks if the site is using a custom domain.
+ */
+function is_custom_domain() {
+	$domain = get_domain();
+	return ! str_ends_with( $domain, '.wpcomstaging.com' ) && ! str_ends_with( $domain, '.wordpress.com' );
 }
 
 /**
  * Decides whether to render to the email nag.
  */
 function domain_email_nag() {
-	if ( ! should_show_domain_frontend_email_nag() ) {
+	if ( ! is_on_frontend_and_logged_in() && ! is_custom_domain() ) {
 		return;
 	}
 
-	$domain = get_domain_with_unverified_email();
+	wp_enqueue_style( 'wpcom-domain-email-nag-style', plugins_url( 'domain-nag.style.css', __FILE__ ), array(), Jetpack_Mu_Wpcom::PACKAGE_VERSION );
+
+	$blog_id = \Jetpack_Options::get_option( 'id' );
 
-	if ( ! $domain ) {
+	if ( ! filter_var( $blog_id, FILTER_VALIDATE_INT ) ) {
 		return;
 	}
 
-	wp_enqueue_style( 'wpcom-domain-email-nag-style', plugins_url( 'domain-nag.style.css', __FILE__ ), array(), Jetpack_Mu_Wpcom::PACKAGE_VERSION );
+	$script = <<<'EOD'
+		const base = 'https://public-api.wordpress.com';
+		const path = '/wpcom/v2/sites/%d/domains/has-unverified-domain-email';
+
+		fetch(base + path).then(function (result) {
+			if (result) {
+				result.json().then(function (body) {
+					if (body.unverified) {
+						const nag = document.querySelector('.wp-domain-nag-sticky-message');
+						console.log( nag );
+						if (nag) {
+							nag.style.display = 'block';
+							const statUrl =
+								'https://pixel.wp.com/b.gif?v=wpcom-no-pv&x_wpcom_frontend_unverified_domain_email_nag=shown';
+							fetch(statUrl);
+						}
+					}
+				});
+			}
+		});
+EOD;
+
+	$script = sprintf(
+		$script,
+		$blog_id
+	);
+
+	$domain = get_domain();
 
 	$notice = sprintf(
-	/* translators: %1 User's email address, %2 current domain */
+		/* translators: %1 User's email address, %2 current domain */
 		__( 'You need to confirm your domain email address to avoid having your domain <strong>%1$s</strong> suspended. Please check your inbox.', 'jetpack-mu-wpcom' ),
 		$domain
 	);
 
 	?>
-	<div class="wp-domain-nag-sticky-message">
+	<div class="wp-domain-nag-sticky-message" style="display:none;">
 		<div class="wp-domain-nag-inner">
 			<p class="wp-domain-nag-text"><?php echo wp_kses( $notice, array( 'strong' => array() ) ); ?></p>
 			<a class="button" href="<?php echo esc_url( get_account_url() ); ?>"><?php esc_html_e( 'Fix', 'jetpack-mu-wpcom' ); ?></a>
 		</div>
 	</div>
+	<script><?php echo $script; // phpcs:ignore -- output generated on server ?></script>
```

Comment:
I like this. Sticking a `<script>` right here is just simpler. I think we can add the `defer` attribute so the JS isn't render blocking and it behaves more like the script appearing in the footer.